### ISSUE TITLE
MongoDB Scaler: ScaledObject give error when we specify port field value without database name.

### ISSUE THREAD

**Issue by Shubham82** [kedacore/keda#1776]
### Report

When i have tried MongoDB scalar with the following ScaledObject yaml:
```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: mongodb-scaledobject
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: nginx-deployment
    envSourceContainerName: nginx
  pollingInterval: 30
  cooldownPeriod:  60
  minReplicaCount: 0
  maxReplicaCount: 100
  advanced:
    restoreToOriginalReplicaCount: true
  triggers:
  - type: mongodb
    metadata:
     host: mongodb-svc.default.svc.cluster.local
     port: "27017"
     username: test_user
     dbName: "test"
     collection: "test_collection"
     query: '{"region":"eu-1","state":"running","plan":"planA"}'
     queryValue: "1"
     metricName: "global-metric"
    authenticationRef:
     name: mongodb-trigger
```
The ScaledObject is created with False status:
```
$ kubectl get so
NAME                   SCALETARGETKIND      SCALETARGETNAME    MIN   MAX   TRIGGERS   AUTHENTICATION    READY   ACTIVE    AGE
mongodb-scaledobject   apps/v1.Deployment   nginx-deployment   0     100   mongodb    mongodb-trigger   False   Unknown   7s
```
 And SO show following Warning:
`failed to ping mongoDB, because of connection() error occured during connection handshake: auth error: sasl conversation error: unable to authenticate using mechanism "SCRAM-SHA-1": (AuthenticationFailed) Authentication failed.`

But when i have changed the port: "27017" to port: "27017/test" where test is the database it is working fine. scaling up and scaling down is working. Here is changed ScaledObject yaml:
```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: mongodb-scaledobject
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: nginx-deployment
    envSourceContainerName: nginx
  pollingInterval: 30
  cooldownPeriod:  60
  minReplicaCount: 0
  maxReplicaCount: 100
  advanced:
    restoreToOriginalReplicaCount: true
  triggers:
  - type: mongodb
    metadata:
     host: mongodb-svc.default.svc.cluster.local
     port: "27017/test"
     username: test_user
     dbName: "test"
     collection: "test_collection"
     query: '{"region":"eu-1","state":"running","plan":"planA"}'
     queryValue: "1"
     metricName: "global-metric"
    authenticationRef:
     name: mongodb-trigger
```
Here is the Secrets, TriggerAuthentication and Deployment yaml which i have used:
```
apiVersion: v1
kind: Secret
metadata:
  name: mongodb-secret
type: Opaque
data:
  connect: dGVzdF9wYXNzd29yZA==
```
```
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: mongodb-trigger
spec:
  secretTargetRef:
    - parameter: password
      name: mongodb-secret
      key: connect
```
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.14.2
        ports:
        - containerPort: 80
```

### Expected Behavior

The ScaledObject should work for `port:"27017"` and take database name from dbName field.

### Actual Behavior

The scaledObject work when we give database name with port value `port:"27017/test"`  in case when we use host,port method (detailed information method)

### Steps to Reproduce the Problem

1. Set-up minikube v1.18.1
2. Deploy mongodb.
3. Create target deployment with above yaml and apply it.
4. Create secrets and TA with above yaml and apply it.
5. Create SO with above yaml's and apply it.
6. Check SO events by kubectl describe so 
7. Check the logs of keda-operator pod by kubectl logs <keda-operator pod name> -n keda

### Logs from KEDA operator

```
{"reconcilerGroup": "keda.sh", "reconcilerKind": "ScaledObject", "controller": "scaledobject", "name": "mongodb-scaledobject", "namespace": "default", "error": "error getting scaler for trigger #0: failed to ping mongoDB, because of connection() error occured during connection handshake
```

### KEDA Version

2.2.0

### Kubernetes Version

1.20

### Scaler Details

MongoDB

**Comment by Shubham82** [comment]
I think we should add [here](https://github.com/kedacore/keda/blob/89da00b24fa961b0c2d58687b031d25cf26e3ce3/pkg/scalers/mongo_scaler.go#L181)

**Comment by Shubham82** [comment]
Hi @tomkerkhove @zroubalik 

Could you please take a look!

**Comment by zroubalik** [comment]
@Shubham82 good catch, you are right. I think that the whole scaler (the input parameters) would really use refactoring on both code and docs side.

Would be great if you can open PRs and we can discuss the changes there, thanks!

**Comment by Shubham82** [comment]
Thanks @zroubalik  for the response.
Will soon create a PR for it.

**Comment by Shubham82** [comment]
Fixed it here #1787
